shell/common/rasterizer.cc

```diff
@@ -270,13 +271,23 @@ sk_sp<SkImage> Rasterizer::DoMakeRasterSnapshot(
                 return;
               }
 
+              GrDirectContext* context = surface_->GetContext();
+
+              // Upper-bound to the max render target size supported by the GPU.
+              // Prefer to proceed with a smaller snapshot over failing with a
+              // null result.
+              auto maxSize = context->maxRenderTargetSize();
+              image_info =
+                  image_info.makeWH(std::min(maxSize, image_info.width()),
+                                    std::min(maxSize, image_info.height()));
+
               // When there is an on screen surface, we need a render target
               // SkSurface because we want to access texture backed images.
-              sk_sp<SkSurface> surface = SkSurface::MakeRenderTarget(
-                  surface_->GetContext(),  // context
-                  SkBudgeted::kNo,         // budgeted
-                  image_info               // image info
-              );
+              sk_sp<SkSurface> surface =
+                  SkSurface::MakeRenderTarget(context,          // context
```

Comment:
This needs a test. IMO, the easiest would be in embedder unittests as that is emulated in an OpenGL test harness. You can get the max size from the host context and ask the Dart code to snapshot a scene larger than that and then check that the size was clamped appropriately.

Reply:
Oh perfect -- I hadn't toured the tests enough to realize. It's nice that scaffolding a test for this kind of thing is easy, kudos for the embedder!

Let me know if I used appropriate communication primitives for i/o with the dart runtime here.